tests/test_store.py

```diff
@@ -124,3 +125,36 @@ async def test_version(YStore, ystore_api, caplog):
         YStore.version = prev_version
         async with ystore as ystore:
             await ystore.write(b"bar")
+
+
+@pytest.mark.parametrize("ystore_api", ("ystore_context_manager", "ystore_start_stop"))
+async def test_compression_callbacks_zlib(ystore_api):
+    """
+    Verify that registering zlib.compress/decompress as callbacks
+    correctly round-trips data through the SQLiteYStore.
+    """
+    async with create_task_group() as tg:
+        store_name = f"compress_test_with_api_{ystore_api}"
+        ystore = MySQLiteYStore(store_name, metadata_callback=MetadataCallback(), delete=True)
+        if ystore_api == "ystore_start_stop":
+            ystore = StartStopContextManager(ystore, tg)
+
+        async with ystore as ystore:
+            # register zlib compression/decompression
+            ystore.register_compression_callbacks(zlib.compress, zlib.decompress)
```

Comment:
Maybe register a no-op for decompression, and decompress in the checks?
```suggestion
            ystore.register_compression_callbacks(zlib.compress, lambda x: x)
```